finetune/input_pipeline.py

```diff
@@ -12,18 +13,37 @@
 from sklearn.model_selection import train_test_split
 
 
+import finetune
 from finetune.errors import FinetuneError
 from finetune.config import PAD_TOKEN
-from finetune.encoding import TextEncoder, ArrayEncodedOutput, EncodedOutput
+from finetune.base_models.gpt.encoder import GPTEncoder
+from finetune.base_models.gpt2.encoder import GPT2Encoder
+from finetune.encoding import ArrayEncodedOutput, EncodedOutput
 from finetune.imbalance import compute_class_weights
 
-ENCODER = TextEncoder()
 LOGGER = logging.getLogger('finetune')
 
+ENCODERS = {
+    'gpt': GPTEncoder,
+    'gpt2': GPT2Encoder
+}
 
 class BasePipeline(metaclass=ABCMeta):
     def __init__(self, config):
         self.config = config
+
+        finetune_base_folder = os.path.dirname(finetune.__file__)
+        self.base_model_root = self.config.base_model_path.split('/')[0]
+        self._encoder = ENCODERS[self.base_model_root]
+        self.text_encoder = self._encoder(
```

Comment:
I think to be generic, this will have to take just the folder and then the encoder can load the files it requires from it.

Reply:
Switch over to method base_model()